### PR TITLE
[WIP] Fixes #25811 - replace multiple_selects by Patternfly's Dual List 

### PR DESCRIPTION
**Pull request by Ron-Lavi**
Replacing the old implementation of `multiple_selects` with Patternfly's Dual list,
and removing the `jquery.multi-select` dependency.

Selection_062

TODO:
- [x] Merge #6720
- [x] Merge patternfly/patternfly-react#1900
- [x] Merge patternfly/patternfly-react#1901
- [x] Merge patternfly/patternfly-react#1909
- [x] Merge patternfly/patternfly-react#1911
- [x] Merge patternfly/patternfly-react#1930
- [x] Remove jquery.multi-select
- [x] refactor `multiple_selects` in form helper
- [x] create storybook
- [ ] get sort icon to appear as in the storybook
- [ ] test patternfly upgrade when all blocking PRs will get merge
- [x] create js tests
- [ ] fix ruby tests

### CODE REVIEW

**Comment by Laviro** [comment]
blocked by https://github.com/patternfly/patternfly-react/pull/1900

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -7,7 +7,6 @@
 //= require patternfly
 //= require vendor
 //= require jquery.extentions
-//= require jquery.multi-select
```
Comment (ohadlevy) [inline app/assets/javascripts/application.js:10]: the actual file (app/assets/javascripts/jquery.multi-select.js) should also be removed.

**Comment by Ron-Lavi** [comment]
[test foreman]

**Comment by mmoll** [comment]
@Laviro you need to fix the Rubocop violation first :)

File: webpack/assets/javascripts/react_app/components/DualList/DualList.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+  render() {
+    const { inputProps, items, selectedIDs, label, id } = this.props;
+    const { selectedList, unselectedlist } = arrangeItemsBySelectedIDs(
+      items,
+      selectedIDs
+    );
+
+    return (
+      <FormGroup id={id}>
```
Comment (ezr-ondrej) [inline webpack/assets/javascripts/react_app/components/DualList/DualList.js]: The #6677 is merged, would it make sense to use it as a wrapper for the component?
I would love to see it usable from the rails helper.

File: webpack/assets/javascripts/react_app/components/DualList/index.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import * as actions from './DualListActions';
+import reducer from './DualListReducer';
+import DualList from './DualList';
+
+const mapStateToProps = (state, { id }) => ({});
```
Comment (ezr-ondrej) [inline webpack/assets/javascripts/react_app/components/DualList/index.js:7]: Why do we have a state in redux, if we are not using it to render the component?
I believe it resonates with my thought we do not need to connect the component, isn't it?

**Review by ezr-ondrej — CHANGES_REQUESTED** [review]
I was thinking about generic components being connected and I believe it encourages to put data in the store by component hierarchy, rather than application domain. The store subtree would now looks like `{dualList: {host_organization_ids: ..., medium_operating_system_ids: ...}}`, what doesn't tells about the application domain too much and is not useful for rendering other parts of application.

Do you think it would make sense without connecting the component, to use inputs `name` attribute if it is standalone component in non react environment and handled the state in higher components (like forms), if needed (if react environment)?

File: webpack/assets/javascripts/react_app/components/DualList/index.js (hunk 3 above)
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/DualList/index.js:7]: still investigating it, but I believe the direction should be to connect it to redux,
and than other components on the page could get updated by changes of the Dual List.

**Comment by Ron-Lavi** [comment]
Thanks @ezr-ondrej for your review,
the PR is still a draft though and I think connecting the component to redux
does make sense so other components could depend on the data that was selected by the DualList,

we can list all of the different Dual List reducers under the domain name, that's what you meant?

**Comment by ezr-ondrej** [comment]
Well my point here is, that it is not the DualList alone who should get stored data, but rather let's say a host edit form of host with (id = 4), who is going to have dual list for selecting organizations and locations. That case should be IMHO reflected in redux something like `{hosts: { host_4: { organizations: ['RedHat'], locations: ['Brno', 'Prague'] }}`, but that is hard to achieve from DualList itself, but would be easy to achieve from the form component, whom would be passing `onChange` handler and just getting the value changes.

I know it's WIP, I was just peeking in the progress and sharing thoughts. I can't wait for seeing the component in action! :+1:

File: webpack/assets/javascripts/react_app/components/DualList/index.js (hunk 3 above)
Comment (ohadlevy) [reply webpack/assets/javascripts/react_app/components/DualList/index.js:7]: I would say that by default (e.g. when using it as a rails form helper) it shouldnt be in redux, it should just mimic the input fields so the existing forms continue to work.

however, when using it inside a react page, then the HOC component would probably be connected with actions firing every time the user takes an action...?

**Comment by ohadlevy** [comment]
whats the status of the PR? thanks.

**Comment by Ron-Lavi** [comment]
Had to work on some other stuff,
I will continue to work on it and improve the feature with the community feedback.

**Comment by xprazak2** [comment]
@stejskalleos I know you were looking into the related issue, so this might be relevant for you.

**Pull request closed without merging**